.github/workflows/run_tests.yaml

```diff
@@ -0,0 +1,38 @@
+name: Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+concurrency: test-${{ github.sha }}
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    env:
+      PYTHON_VERSION: "3.10"
+
+    steps:
+      - name: Checks out repository
+        uses: actions/checkout@v3
+
+      - name: Set up Python ${{ env.PYTHON_VERSION }}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pytest anyio
+          if [ -f dev-requirements.txt ]; then pip install -r dev-requirements.txt; fi
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+
+      - name: Setup database container
```

Comment:
i don't think the server has the image for the dev database, however it can just access the dev database if we provide the credentials in a .env file

Reply:
not sure if this can access the repo secrets?

Reply:
Won't know until the workflow can be approved at the bottom. I think that should allow it to run and we can see any errors from there.

Reply:
we can checkout the repo for mysql aswell? the annoying part is we would need to wait for "ready for connections".

folder structure must be
/
/mysql-repo-name
/api-repo-name

Reply:
> do we have dev-requirements?

No. This was just something I copied from another project I developed.